doc/man7/ossl-guide-libcrypto-introduction.pod

```diff
@@ -0,0 +1,398 @@
+which has been validated to FIPS standards for situations where that is
+important. It is also possible that a third party could add additional
+implementations such as in a hardware security module (HSM).
+
+Algorithms are implemented in providers. See
+L<ossl-guide-libraries-introduction(7)> for information about providers.
+
+=head2 Operations
+
+Different algorithms can be grouped together by their purpose. For example there
+are algorithms for encryption, and different algorithms for digesting data.
+These different groups are known as "operations" in OpenSSL. Each operation
+has a different set of functions associated with it. For example to perform an
+encryption operation using AES (or any other encryption algorithm) you would use
+the encryption functions detailed on the L<EVP_EncryptInit(3)> page. Or to
+perform a digest operation using SHA256 then you would use the digesting
+functions on the L<EVP_DigestInit(3)> page.
+
+=head1 ALGORITHM FETCHING
+
+In order to use an algorithm an implementation for it must first be "fetched".
+Fetching is the process of looking through the available implementations,
+applying selection criteria (via a property query string), and finally choosing
+the implementation that will be used.
+
+Two types of fetching are supported by OpenSSL - L</Explicit fetching> and
+L</Implicit fetching>.
+
+=head2 Explicit fetching
+
+Explicit fetching involves directly calling a specific API to fetch an algorithm
+implementation from a provider. This fetched object can then be passed to other
+APIs. These explicit fetching functions usually have the name C<APINAME_fetch>,
+where C<APINAME> is the name of the operation. For example L<EVP_MD_fetch(3)>
+can be used to explicitly fetch a digest algorithm implementation. The user is
+responsible for freeing the object returned from the C<APINAME_fetch> function
+using C<APINAME_free> when it is no longer needed.
+
+These fetching functions follow a fairly common pattern, where three
+arguments are passed:
+
+=over 4
+
+=item The library context
+
+See L<OSSL_LIB_CTX(3)> for a more detailed description.
+This may be NULL to signify the default (global) library context, or a
+context created by the user. Only providers loaded in this library context (see
+L<OSSL_PROVIDER_load(3)>) will be considered by the fetching function. In case
+no provider has been loaded in this library context then the default provider
+will be loaded as a fallback (see L<OSSL_PROVIDER-default(7)>).
+
+=item An identifier
+
+For all currently implemented fetching functions this is the algorithm name.
+Each provider supports a list of algorithm implementations. See the provider
+specific documentation for information on the algorithm implementations
+available in each provider:
+L<OSSL_PROVIDER-default(7)/OPERATIONS AND ALGORITHMS>,
+L<OSSL_PROVIDER-FIPS(7)/OPERATIONS AND ALGORITHMS>,
+L<OSSL_PROVIDER-legacy(7)/OPERATIONS AND ALGORITHMS> and
+L<OSSL_PROVIDER-base(7)/OPERATIONS AND ALGORITHMS>.
```

Comment:
Maybe supply some nicer text to read for the links?
e.g. [default](OSSL_PROVIDER-default.html#OPERATIONS-AND-ALGORITHMS), [FIPS](OSSL_PROVIDER-FIPS.html#OPERATIONS-AND-ALGORITHMS), [legacy](OSSL_PROVIDER-legacy.html#OPERATIONS-AND-ALGORITHMS) and [base](OSSL_PROVIDER-base.html#OPERATIONS-AND-ALGORITHMS).